terminus-ui/logo/src/logo.component.ts

```diff
@@ -0,0 +1,163 @@
+  'white',
+  'black',
+  'gray',
+];
+
+
+/**
+ * This is the logo UI Component
+ *
+ * #### QA CSS CLASSES
+ * - `qa-label`: Placed on the primary container
+ *
+ * @example
+ * <ts-logo type="full-gradient"></ts-logo>
+ * <ts-logo type="solid-mark" logoColor="gray"></ts-logo>
+ *
+ * <example-url>https://getterminus.github.io/ui-demos-master/components/logo</example-url>
+ */
+@Component({
+  selector: 'ts-logo',
+  templateUrl: './logo.component.html',
+  styleUrls: ['./logo.component.scss'],
+  host: {
+    class: 'ts-logo',
+    '[class.ts-logo--black]': 'logoColor === "black"',
+    '[class.ts-logo--gray]': 'logoColor === "gray"',
+  },
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  encapsulation: ViewEncapsulation.None,
+  exportAs: 'tsLogo',
+})
+
+export class TsLogoComponent implements OnInit {
+  /**
+   * Name of the logo types
+   */
+  @Input()
+  public set type(value: TS_LOGO_TYPE) {
+    // If an unsupported value is passed in
+    if (value && TS_LOGO_TYPES.indexOf(value) < 0 && isDevMode()) {
+      console.warn(`TsLogoComponent: "${value}" is not a supported logo type. ` +
+      `See TS_LOGO_TYPE for available options.`);
+      return;
+    }
+    this._type = value;
+  }
+  public get type(): TS_LOGO_TYPE {
+    return this._type;
+  }
+  private _type: TS_LOGO_TYPE = 'full-gradient';
+
+
+  /**
+   * Name of the logo colors
+   */
+  @Input()
+  public set logoColor(value: TS_LOGO_COLOR) {
+    // If an unsupported value is passed in
+    if (value && TS_LOGO_COLORS.indexOf(value) < 0 && isDevMode()) {
+      console.warn(`TsLogoComponent: "${value}" is not a supported logo color. ` +
+      `See TS_LOGO_COLOR for available options.`);
+      return;
+    }
+    this._logoColor = value;
+  }
+  public get logoColor(): TS_LOGO_COLOR {
+    return this._logoColor;
+  }
+  private _logoColor: TS_LOGO_COLOR = 'white';
+
+
+  public svg: SafeHtml = ``;
+
+
+  constructor(
+    private domSanitizer: DomSanitizer,
+  ) {}
+
+  private setType() {
+    switch (this.type) {
+      case('full-account-hub'):
+        this.logoColor = 'white';
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(FULL_ACCOUNT_HUB_LOGO);
+        break;
+      case('full-gradient'):
+        this.logoColor = 'white';
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(FULL_GRADIENT_LOGO);
+        break;
+      case('full-solid'):
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(FULL_SOLID_LOGO);
+        break;
+      case('mark-gradient'):
+        this.logoColor = 'white';
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(MARK_GRADIENT_LOGO);
+        break;
+      case('mark-solid'):
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(MARK_SOLID_LOGO);
+        break;
+      default:
+        this._logoColor = 'white';
+        this.svg = this.domSanitizer.bypassSecurityTrustHtml(FULL_GRADIENT_LOGO);
+        break;
+    }
+  }
+
+  public ngOnInit(): void {
```

Comment:
We lost a comment here..

Let's move this above `setType`.

In general our order is usually:

```
properties
getters
inputs
outputs
constructor
lifecycle events
public methods
private methods
````